File: code/python/sumEvenAfterQueries.py
class Solution:
    def sumEvenAfterQueries(self, A: List[int], queries: List[List[int]]) -> List[int]:
        if len(queries) == 0:
            return [];
        res = [0 for i in range(len(queries))]
        A[queries[0][1]] += queries[0][0]
        count = 0
        
        for i in range(len(A)):
            if A[i] % 2 == 0:
                count += A[i]
        res[0] = count

        for i in range(1,len(queries)):
            prev = A[queries[i][1]]
            A[queries[i][1]] += queries[i][0]
            if A[queries[i][1]] % 2 == 0 and prev % 2 == 0:
                count = count + queries[i][0]
            elif A[queries[i][1]] % 2 == 0 and prev % 2 != 0:
                count += A[queries[i][1]]
            elif A[queries[i][1]] % 2 != 0 and prev % 2 == 0:
                count -= prev
            res[i] = count

        return res

File: code/java/sumEvenAfterQueries.java
class Solution {
    public int[] sumEvenAfterQueries(int[] A, int[][] queries) {
        if(queries.length == 0)
            return null;
        int[] res = new int[queries.length];
        A[queries[0][1]] += queries[0][0];
        for(int i=0;i<A.length;i++)
            if(A[i] % 2 == 0)
                res[0] += A[i];
        for(int i=1;i<queries.length;i++){
            int prev = A[queries[i][1]];
            A[queries[i][1]] += queries[i][0];
            if(A[queries[i][1]]%2 == 0)
                if(prev % 2 == 0)
                    res[i] = res[i-1] + queries[i][0];
                else
                    res[i] = res[i-1] + A[queries[i][1]];
            else
                if(prev % 2 == 0)
                    res[i] = res[i-1] - prev;
                else
                    res[i] = res[i-1];
        }
        
        return res;
    }
}
